sg_covid_impact/claimant_count.py

```diff
@@ -1,45 +1,59 @@
 
+_CLAIM_FILE = f"{project_dir}/data/processed/claimant_counts.csv"
+
+def load_claimant_data(loc):
+    """Loads claimant data from NOMIS or locally
+    Args:
+        loc (str): location of claimant file
+
+    """
+    return pd.read_csv(loc)
+
+def process_claimant_data(table):
+    """Processes claimant data
+    Args:
+        table (df): dataframe with raw claimant data
+    """
+    table.columns = [x.lower() for x in table.columns]
+    table["date"] = pd.to_datetime(table["date"], format="%Y-%m")
+
+    rel_vars = ["date", "geography_name", "geography_code", "measure_name", "obs_value"]
+
+    return table[rel_vars]
 
 def make_claimant_data():
     """Collect and process claimant data.
     Returns the processed dataset and the last month in the data
     """
     # Collect data
-    claimants_raw = pd.read_csv(_CLAIM_URL)
+    claimants_raw = load_claimant_data(_CLAIM_URL)
     # Process
-    claimants_raw.columns = [x.lower() for x in claimants_raw.columns]
-    claimants_raw["date_full"] = pd.to_datetime(
-        [x + "-01" for x in claimants_raw["date"]]
-    )
-
-    rel_vars = [
-        "date",
-        "date_name",
-        "geography_name",
-        "geography_code",
-        "measure_name",
-        "obs_value",
-        "date_full",
-    ]
-    claimants = claimants_raw[rel_vars]
-    last_month = str(claimants["date_full"].max().month)
-
-    return last_month, claimants
+    claimant_proc = process_claimant_data(claimants_raw)
+
+    last_month = str(claimant_proc["date"].max().month)
+
+    if os.path.exists(_CLAIM_FILE):
+        # If we already collected before we load the old file,
+        # concatenate and drop duplicated date-area rows
```

Comment:
This strategy doesn't persist across computers.
We could push and pull from S3 but a simpler solution is to change the end of the NOMIS query to `...&date=2019-12-latest&gender=0&age=0&measure=1,2&measures=20100`, thus ensuring that the start date of the dataset stays consistent over time.